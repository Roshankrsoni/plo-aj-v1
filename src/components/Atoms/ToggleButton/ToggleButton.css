.ToggleButton {
  position: relative;
  width: 36px;
  height: 1rem;
  display: flex;
  align-items: center;
}

.circle {
  width: 17px;
  height: 17px;

  border-radius: 50%;

  position: absolute;
  top: 50%;
  transform: translateY(-50%);

  transition: all 0.3s ease-in;
}

.bar {
  height: 20px;
  width: 100%;
  border-radius: 50px;
  padding: 2px;
}

#on .bar {
  background-color: var(--light-orange);
}
#off .bar {
  background-color: var(--grey5);
}
#on .circle {
  background-color: var(--red);
  left: calc(100% - 19px);
}
#off .circle {
  background-color: var(--grey4);
  left: 2px;
}
