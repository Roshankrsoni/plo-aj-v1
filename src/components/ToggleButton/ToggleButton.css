.ToggleButton{
    position: relative;
    width: 36px;
    height: 1rem;
    display: flex;
    align-items: center;
}

.circle{
    width:1rem;
    height:1rem;
  
    border-radius: 50%;

    position: absolute;
    top:50%;
    transform: translateY(-50%);

    transition: all 0.3s ease-in;
}

.bar{
    height:5px;
  
    width:36px;
    border-radius:10px;
}

#on .bar{
    background-color: #FFAEAE;
}
#off .bar{
    background-color: #E0E0E0;
}
#on .circle{
    background-color: #EB5757;
    left:calc(100% - 1rem);
}
#off .circle{
    background-color: #BDBDBD;
    left:0;
}


