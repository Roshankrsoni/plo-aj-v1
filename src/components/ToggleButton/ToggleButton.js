import React from 'react'
import './ToggleButton.css'

const ToggleButton = (props) => {
    function clickHandler(evt){
        props.onClick();
    } 
    return (
        <div id={props.value ? 'on' : 'off'} className="ToggleButton" onClick={clickHandler}>
          <div className="bar"></div>
          <div className="circle"></div>
        </div>
    )
}

export default ToggleButton;