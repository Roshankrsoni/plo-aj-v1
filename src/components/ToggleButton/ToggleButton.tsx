import React from "react";
import "./ToggleButton.css";
interface ToggleButtonProps {
  onClick: () => void;
  value: boolean;
  trueLabel: string;
  falseLabel: string;
}
const ToggleButton = ({ onClick, value }: ToggleButtonProps) => {
  function clickHandler() {
    onClick();
  }
  return (
    <div
      id={value ? "on" : "off"}
      className="ToggleButton"
      onClick={clickHandler}
    >
      <div className="bar">
        <div className="circle"></div>
      </div>
    </div>
  );
};

export default ToggleButton;
