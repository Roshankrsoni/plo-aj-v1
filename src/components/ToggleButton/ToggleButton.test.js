
import React from 'react';
import { shallow } from 'enzyme';
import ToggleButton from './ToggleButton';

describe('MyComponent', () => {
    it('should render correctly in "debug" mode', () => {
        const component = shallow(<ToggleButton />);
        expect(component).toMatchSnapshot();
    });
});