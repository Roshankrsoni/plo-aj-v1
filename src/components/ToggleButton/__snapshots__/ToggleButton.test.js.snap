// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render correctly in "debug" mode 1`] = `
<div
  className="ToggleButton"
  id="off"
  onClick={[Function]}
>
  <div
    className="bar"
  >
    <div
      className="circle"
    />
  </div>
</div>
`;
